src/SynchronizesWithMauticTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Swis\Laravel\Mautic;
+
+use Swis\Laravel\Mautic\Facades\Mautic;
+
+trait SynchronizesWithMauticTrait
+{
+    public static function bootSynchronizesWithMauticTrait()
+    {
+        static::saving(function ($model) {
+            if ($model instanceof SynchronizesWithMautic) {
+                $mautic = Mautic::connection($model->getMauticConnection());
+                if ($model->getMauticId()) {
+                    call_user_func([$mautic, $model->getMauticType()])
+                        ->edit($model->getMauticId(), $model->toMauticArray());
+
+                    return;
+                }
+                $response = call_user_func([$mautic, $model->getMauticType()])
+                    ->create($model->toMauticArray());
+
+                $model->setMauticId($response['contact']['id']);
```

Comment:
This isn't always 'contact', it can also be 'company', etc. You can get the correct name using `call_user_func([$mautic, $model->getMauticType()])->itemName()`.